concordium-consensus/src/Concordium/GlobalState/Persistent/BlobStore.hs

```diff
@@ -290,12 +297,25 @@ truncateBlobStore BlobStoreAccess{..} (BlobRef offset) = do
     case esize :: Either String Word64 of
       Right size -> do
         let newSize = offset + 8 + size
-        hSetFileSize bhHandle $ fromIntegral newSize
+        unless (bhSize == fromIntegral newSize) $ do
+          -- unmap the current memory mapped file since on some platforms the file cannot be
+          -- truncated if it is memory mapped.
+          oldMmap <- readIORef blobStoreMMap
+          -- write a temporary empty string.
+          writeIORef blobStoreMMap BS.empty
+          let (fp, _, _) = BSInternal.toForeignPtr oldMmap
+          -- unmap the old mapping.
+          finalizeForeignPtr fp
```

Comment:
A matter of taste and no deal breaker, but I think it could be nice to have this factored out in a separate `where` func decl. `unmap` in order to make keep the abstraction level the same throughout the internals of the function.

Reply:
Fair comment, but I think I prefer the way it is.